src/lib.rs

```diff
@@ -38,13 +38,37 @@ struct Package {
 }
 
 /// Get the repository name from `Cargo.toml`.
+/// Fallbacks to directory name if `Cargo.toml` does not exists.
 pub fn read_repo(dir: &str) -> ::Result<String> {
   let mut dir = PathBuf::from(dir);
   dir.push("Cargo.toml");
-  let cargo_toml = fs::read_to_string(dir).context(::ErrorKind::Other)?;
-  let config: Config =
-    toml::from_str(&cargo_toml).context(::ErrorKind::Other)?;
-  Ok(config.package.repository)
+
+  if dir.as_path().exists() {
+    let cargo_toml = fs::read_to_string(dir).context(::ErrorKind::Other)?;
+    let config: Config =
+      toml::from_str(&cargo_toml).context(::ErrorKind::Other)?;
+    Ok(config.package.repository)
+  } else {
+    dir.pop();
+    match dir.file_name() {
+      Some(file_name) => {
+        let config: Config = Config {
+          package: Package {
+            repository: file_name.to_str().unwrap().to_string(),
+          },
+        };
+        Ok(config.package.repository)
+      }
+      None => {
+        let config: Config = Config {
+          package: Package {
+            repository: String::from(""),
+          },
+        };
+        Ok(config.package.repository)
```

Comment:
Every branch of the `if/else/match` statement returns a `Ok(config.package.repository)`. A cleaner way to write it might be to return the `Config` from the statement, and then return `Ok(config.package.repository)` once at the end.


```rust
let config = if dir.as_path.exists() {
 // ...
} else {
  dir.pop();
  match dir.file_name() {
    // ...
  }
};

Ok(config.package.repository)
```